src/EditorFeatures/Core.Wpf/Suggestions/SuggestedActionsSource.cs

```diff
@@ -408,36 +370,34 @@ private static string GetFixCategory(DiagnosticSeverity severity)
                 TextSpan? selection = null;
                 if (IsForeground())
                 {
-                    selection = TryGetCodeRefactoringSelection(range);
+                    selection = TryGetCodeRefactoringSelection(state, range);
                 }
                 else
                 {
                     await InvokeBelowInputPriorityAsync(() =>
                     {
-                        // Make sure we were not disposed between kicking off this work and getting
-                        // to this point.
-                        if (IsDisposed)
-                        {
+                        // Make sure we were not disposed between kicking off this work and getting to this point.
+                        using var state = _state.TryAddReference();
```

Comment:
it's esp confusing, now that we have `state` above and `state` here.

Reply:
This is intentional hiding (the outer `state` is not valid in this block).

Reply:
oh i got that.  it didn't make it less confusing :)